src/models/Util.ts

```diff
@@ -78,6 +80,33 @@ export const Util = {
 
     isTechnicalCivilisation(civilisation: Civilisation): boolean {
         return civilisation.gameVersion === GameVersion.TECHNICAL;
-    }
+    },
 
+    isRandomCivilisation(civilisation: Civilisation): boolean {
+        return civilisation.name.toUpperCase() === "RANDOM";
+    },
+
+    getRandomCivilisation(): Civilisation {
+        const maxCivilisationIndex = Civilisation.ALL.length;
```

Comment:
The Preset could have a different set of available civilisations, for example only civs from AoK/AoC. We should only try civs from the allowed set of civilisations in `draft.preset.civilisations`

Reply:
We could start off with a copy of `draft.preset.civilisations` in `setRandomCivilisationIfNeeded()`, and then remove any civ that we tried. This way we do not have to check a civilisation twice that we already rejected before.